rust/operator-binary/src/catalog/commons.rs

```diff
@@ -91,22 +93,53 @@ impl ExtendCatalogConfig for S3ConnectionInlineOrReference {
             .await
             .context(ConfigureS3Snafu)?;
 
-        catalog_config.add_property("hive.s3.endpoint", s3.endpoint().context(ConfigureS3Snafu)?);
-        catalog_config.add_property(
-            "hive.s3.path-style-access",
-            (s3.access_style == S3AccessStyle::Path).to_string(),
-        );
-
         let (volumes, mounts) = s3.volumes_and_mounts().context(ConfigureS3Snafu)?;
         catalog_config.volumes.extend(volumes);
         catalog_config.volume_mounts.extend(mounts);
 
+        if trino_version >= 469 {
+            // Since Trino 469, S3 native support has to be explicitly enabled
+            // uness using Legacy S3 support (which has been removed in 470).
+            catalog_config.add_property("fs.native-s3.enabled", "true");
+
+            // Trino silently fails to load if `s3.region` is not set.
+            // We will hard-code it for now so that Trino can work with IONOS
+            // and Minio, but this will eventually come in via `S3Connection`
+            // once the CRD has the field.
+            tracing::warn!("s3.region is currently hard-coded to us-east-1");
+            catalog_config.add_property("s3.region", "us-east-1");
```

Comment:
> [!CAUTION]
> We either need to merge with this in, or fix the CRD now to handle the s3 region.

Reply:
I would be in favor of adding the region to the S3Connection (via a decision). We need to think about defaults though

Reply:
Link to decision: https://github.com/stackabletech/decisions/issues/46#issuecomment-2668568876

Reply:
Done in bc54a712da2b2b6d580fa184b538ae3c68048dfd

This was [decided](https://github.com/stackabletech/decisions/issues/46) and implemented in [stackable-operator](https://github.com/stackabletech/operator-rs/pull/959) and [commons-operator](https://github.com/stackabletech/commons-operator/pull/331).

Reply:
Thanks!